### PR TITLE
Typed testkit: Adding proper NoEffects assertions

### LINKED ISSUE #25061: Testkit typed: NoEffects expectations are broken.
Hello! 
I am trying to test actor that performs no actions at all.

Actor:
```
val init = Behaviors.receiveMessage[Int] {
  case _ => Behavior.same
}
```

And test:
```
val testkit = BehaviorTestKit[Int](init)
testkit.expectEffect(NoEffects)
```
As a test result a got some strange assertion failure: 

> expected: NoEffects but no effects were recorded.

This is looks surprising, because scaladoc for NoEffects says, that this effect means that behavior didn't do anything.

### PR DESCRIPTION
**Pull request by ilyshav**
fix #25061

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

File: akka-testkit-typed/src/test/scala/akka/testkit/typed/scaladsl/BehaviorTestKitSpec.scala [hunk 1]
```diff
@@ -114,6 +114,16 @@ class BehaviorTestKitSpec extends WordSpec with Matchers {
       spawnAnonymous.props should ===(Props.empty)
     }
 
+    "allow expecting NoEffects by type" in {
+      val testkit = BehaviorTestKit[Father.Command](Father.init)
+      testkit.expectEffectType[NoEffects.type]
```
Comment (johanandren) [inline akka-testkit-typed/src/test/scala/akka/testkit/typed/scaladsl/BehaviorTestKitSpec.scala]: That's somewhat weird, and I guess we want people to use the other syntax?

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-testkit-typed/src/test/scala/akka/testkit/typed/scaladsl/BehaviorTestKitSpec.scala (hunk 1 above)
Comment (patriknw) [reply akka-testkit-typed/src/test/scala/akka/testkit/typed/scaladsl/BehaviorTestKitSpec.scala]: could we use something like this:
```
  trait NoEffects extends Effect
  
  case object NoEffects extends NoEffects
```

Comment (ilyshav) [reply akka-testkit-typed/src/test/scala/akka/testkit/typed/scaladsl/BehaviorTestKitSpec.scala]: That's much better, thank you. I did it.

**Review by chbatey — APPROVED** [review]
LGTM

**Pull request merged**